src/errors/checkXcmTxInputs.ts

```diff
@@ -380,22 +552,145 @@ const checkParaToSystemAssetId = (
+ *
+ * @param assetIds
+ */
+export const checkAssetIdsLengthIsValid = (assetIds: string[]) => {
+	if (assetIds.length > MAX_ASSETS_FOR_TRANSFER) {
+		throw new BaseError(
+			`Maximum number of assets allowed for transfer is 2. Found ${assetIds.length} assetIds`
+		);
+	}
+};
+
+/**
+ * Checks to ensure that assetId's have no duplicates, throws an error if duplicate is found
+ *
+ * @param assetIds
+ */
+export const checkAssetIdsHaveNoDuplicates = (assetIds: string[]) => {
+	const duplicateAssetIds: string[] = [];
+
+	if (assetIds.length > 1) {
+		for (let i = 0; i < assetIds.length; i++) {
+			const asset1 = assetIds[i];
+
+			for (let j = 0; j < assetIds.length; j++) {
+				if (i === j) {
+					continue;
+				}
+
+				const asset2 = assetIds[j];
+				if (
+					asset1.trim().toLowerCase().replace(/ /g, '') ===
+					asset2.trim().toLowerCase().replace(/ /g, '')
+				) {
+					duplicateAssetIds.push(asset2);
+				}
+			}
+		}
+
+		if (duplicateAssetIds.length > 0) {
+			if (assetIds[0] === '') {
+				throw new BaseError(
+					`AssetIds must be unique. Found duplicate native relay assets as empty strings`
+				);
+			}
+
+			throw new BaseError(
+				`AssetIds must be unique. Found duplicate assetId ${duplicateAssetIds[0]}`
+			);
+		}
+	}
+};
+
+/**
+ * Checks to ensure that assetId's are symbols, integers and empty string or multilocations exclusively
+ *
+ * @param assetIds
+ */
+export const checkAssetIdsAreOfSameAssetIdType = (assetIds: string[]) => {
+	if (assetIds.length > 1) {
+		let relayDefaultValueFound = false;
+		let symbolAssetIdFound = '';
+		let integerAssetIdFound = '';
+		let multiLocationAssetIdFound = '';
+
+		for (const assetId of assetIds) {
+			if (assetId === '') {
+				relayDefaultValueFound = true;
+				continue;
+			}
+
+			const parsedAssetIdAsNumber = Number.parseInt(assetId);
+			const isNotANumber = Number.isNaN(parsedAssetIdAsNumber);
+
+			if (!isNotANumber) {
+				integerAssetIdFound = assetId;
+			} else if (assetId.toLowerCase().includes('parents')) {
+				multiLocationAssetIdFound = assetId;
+			} else {
+				symbolAssetIdFound = assetId;
+			}
+		}
+
+		if (relayDefaultValueFound && multiLocationAssetIdFound) {
+			throw new BaseError(
+				`Found both default relay native asset and multilocation assetId: ${multiLocationAssetIdFound}. Asset Ids must be either be a combination of empty string, symbol and integer or multilocations exclusively.`
+			);
+		}
+
+		if (integerAssetIdFound && multiLocationAssetIdFound) {
+			throw new BaseError(
+				`Found both integer ${integerAssetIdFound} and multilocation assetId ${multiLocationAssetIdFound}. Asset Ids must be symbol and integer or multilocation exclusively.`
+			);
+		}
+
+		if (symbolAssetIdFound && multiLocationAssetIdFound) {
+			throw new BaseError(
+				`Found both symbol ${symbolAssetIdFound} and multilocation assetId ${multiLocationAssetIdFound}. Asset Ids must be symbol and integer or multilocation exclusively.`
```

Comment:
is this true in general? what if the registry has also foreign assets by symbol (which I assume should be the case in the future)?

Reply:
the current way we determine if an asset is a foreign asset is if it is passed in as a multilocation. If we expect that symbols will eventually be present in the foreign assets pallet as well then we would need to change our implementation to account for this scenario

Reply:
that is a good point. I expect foreign assets to be identified by symbols in the same way as any other asset. But lets leave this functionality for later. I think it is ok for now to leave it like this